packages/core/src/Account.ts

```diff
@@ -272,15 +271,15 @@ export class Account extends TypedEventEmitter<Events> {
     await this.service.notification.initializeNotificationStream();
     await this.service.client.synchronizeClients();
 
-    await this.initClient(client);
-    return client;
+    return this.initClient(client);
   }
 
   /**
    * Will initiate all the cryptographic material of the device and setup all the background tasks.
    *
    * @returns The local existing client or undefined if the client does not exist or is not valid (non existing on backend)
    */
+  public async initClient(client: RegisteredClient): Promise<RegisteredClient>;
   public async initClient(client?: RegisteredClient): Promise<RegisteredClient | undefined> {
```

Comment:
This improves the return type of the function (if a client is given, then the return type is not going to be undefined)